OpenLaTeXStudio/EditorModule/src/latexstudio/editor/files/FileChooserService.java

```diff
@@ -63,7 +64,23 @@ public static File getSelectedFile(String extension, String description, DialogT
             File file = chooser.getSelectedFile();
             if (!file.exists() && type == DialogType.OPEN) {
                 JOptionPane.showMessageDialog(null, "File doesn't exist", "Item not found", JOptionPane.ERROR_MESSAGE); 
-            } else {              
+            } 
+            else if(file.exists() && type == DialogType.SAVE) {
+                int promptResponse = showOverwritePrompt(file);
+                while(promptResponse == JOptionPane.NO_OPTION){
+                        chooser.showSaveDialog(null);
+                        file = chooser.getSelectedFile();
+                        if(file.exists()) { 
+                            promptResponse = showOverwritePrompt(file);
+                        }
+                }
+                String filePath = file.getAbsolutePath();

```

Comment:
This part that adds extension to the file, in case it's missing, should be moved higher. Currently we might have file named "welcome.tex" and save file named "welcome" - there will be no overwrite prompt (the file names are different) but the file will be overwritten anyways, since it will add ".tex" here